### ISSUE TITLE
3.8 and 3.9 don't build

### ISSUE THREAD

**Issue by Chrismarsh** [firelab/windninja#508]
I wanted to bring to the dev's attention that neither 3.8 or 3.9 build for me on macos

```
$ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

One class of error impacting 3.8, 3.9, and master I have addressed in PR #507 regarding the mismatch signatures containing FILE and VSILFILE.

The second problem is that 3.9 also has a broken copy constructor around turbulence not being protected for non-foam builds
https://github.com/firelab/windninja/blob/3.9/src/ninja/ninja.cpp#L107

This appears resolved in `master` though.

**Comment by nwagenbrenner** [comment]
@Chrismarsh I merged your PR to 3.9 and 3.8 and also cherry picked e4009269433ffd340e748bb40f78057a88d83799 to 3.9. Let us know if you run into other issues.

**Comment by Chrismarsh** [comment]
Thanks @nwagenbrenner ! I'm wondering, would it be possible to get a new 3.9 tag? e.g., 3.9.1?

**Comment by nwagenbrenner** [comment]
I just tagged 3.9.0.1. We may do a bug fix release with tag 3.9.1, so used 3.9.0.1 to differentiate this as a non-release tag.
